casr/src/bin/casr-cluster.rs

```diff
@@ -408,6 +600,44 @@ fn main() -> Result<()> {
                     INPUT_DIR will be added to OUTPUT_DIR.",
                 ),
         )
+        .arg(
+            Arg::new("update")
+                .short('u')
+                .long("update")
+                .action(ArgAction::Set)
+                .num_args(2)
+                .value_parser(clap::value_parser!(PathBuf))
+                .value_names(["NEW_DIR", "OLD_DIR"])
+                .help(
+                    "Update clusters from OLD_DIR using CASR reports from NEW_DIR.",
+                ),
+        )
+        .arg(
+            Arg::new("inner-strategy")
+                .long("inner-strategy")
+                .value_name("STRATEGY")
+                .action(ArgAction::Set)
+                .value_parser(["Diam", "Dist"])
+                .default_value("Dist")
+                .help("Strategy for inner cluster choosing when updating"),
+        )
+        .arg(
+            Arg::new("outer-strategy")
+                .long("outer-strategy")
+                .value_name("STRATEGY")
+                .action(ArgAction::Set)
+                .value_parser(["Delta", "Diam", "Dist"])
+                .default_value("Dist")
+                .help("Strategy for outer cluster choosing when updating"),
+        )
+        .arg(
+            Arg::new("estimate")
+                .long("estimate")
+                .value_name("DIR")
+                .action(ArgAction::Set)
+                .value_parser(clap::value_parser!(PathBuf))
+                .help("Make cluster estimation for DIR using silhouette index"),
```

Comment:
Calculate silhouette score for clustering results

Reply:
Done